Lib/test/test_platform.py

```diff
@@ -741,5 +745,73 @@ def test_parse_os_release(self):
         self.assertEqual(len(info["SPECIALS"]), 5)
 
 
+class CommandLineTest(unittest.TestCase):
+    def setUp(self):
+        platform.invalidate_caches()
+        self.addCleanup(platform.invalidate_caches)
+
+    @staticmethod
+    def text_normalize(string):
+        """Dedent *string* and strip it from its surrounding whitespaces.
+
+        This method is used by the other utility functions so that any
+        string to write or to match against can be freely indented.
+        """
+        return dedent(string).strip()
+
+    def invoke_platform(self, *flags):
+        output = io.StringIO()
+        with contextlib.redirect_stdout(output):
+            platform._main(args=flags)
+        return self.text_normalize(output.getvalue())
+
+    def test_unknown_flag(self):
+        with self.assertRaises(SystemExit):
+            # suppress argparse error message
+            with contextlib.redirect_stderr(io.StringIO()):
+                _ = self.invoke_platform('--unknown')
+
+    def test_invocation(self):
+        flags = (
+            "--terse", "--nonaliased", "terse", "nonaliased"
+        )
+
+        for r in range(len(flags) + 1):
+            for combination in itertools.combinations(flags, r):
+                self.invoke_platform(*combination)
+
+    def test_arg_parsing(self):
+        # Due to backwards compatibility, the `aliased` and `terse` parameters
+        # are computed based on a combination of positional arguments and flags.
+        #
+        # This test tests that the arguments are correctly passed to the underlying
+        # `platform.platform()` call. The parameters are two booleans for `aliased`
+        # and `terse`.
+        options = (
+            (["--nonaliased"], (False, False)),
+            (["nonaliased"], (False, False)),
+            (["--terse"], (True, True)),
+            (["terse"], (True, True)),
+            (["nonaliased", "terse"], (False, True)),
+            (["--nonaliased", "terse"], (False, True)),
+            (["--terse", "nonaliased"], (False, True)),
+        )
+
+        for flags, args in options:
+            with self.subTest(flags=flags, args=args):
+                with mock.patch.object(platform, 'platform') as obj:
+                    self.invoke_platform(*flags)
+                    obj.assert_called_once_with(*args)
+
+    def test_help(self):
+        output = io.StringIO()
+
+        with self.assertRaises(SystemExit):
+            with contextlib.redirect_stdout(output):
+                platform._main(args=["--help"])
+
+        self.assertIn("usage:", output.getvalue())
```

Comment:
I would prefer to use `assertStartsWith` and we already have `invoke_platform` method to redirect stdout
```suggestion
        with self.assertRaises(SystemExit):
            output = self.invoke_platform("--help")
            self.assertStartsWith(output, "usage: ")
```

Reply:
I agree with all of your review comments except for using the existing `invoke_platform` method. Since the call to `parse_args(args=["--help"])` raises a SystemExit here, using the `invoke_platform` function here would just raise the Exception and wouldn't return the `stdout`.

I'm not sure there is a good way of having `invoke_platform` catch the exception, and catch the logs as-well, so I just call `platform._main()` directly in this test. Its similar to `test_unknown_flag()` except we're reading `stdout` instead of `stderr` here.

Reply:
sorry, my bad

Reply:
sure, got you. using `assetStartsWith` is enough